services/jvm/src/main/scala/io/gearpump/services/MasterService.scala

```diff
@@ -184,49 +198,124 @@ object MasterService {
+      systemConfig,
+      userConfigFile
+    )
+  }
+
+  /**
+   * Submit Storm application.
+   */
+  def submitStormApp(jar: Option[File], stormConf: Option[File], args: String, systemConfig: Config): Boolean = {
+    submitAndDeleteTempFiles(
+      "io.gearpump.experiments.storm.StormRunner",
+      argsArray = spaceSeparatedArgumentsToArray(args),
+      fileMap = Map("jar" -> jar, "config" -> stormConf).filter(_._2.isDefined).mapValues(_.get),
+      classPath = getStormApplicationClassPath,
+      systemConfig
+    )
+  }
 
+  private def submitAndDeleteTempFiles(mainClass: String, argsArray: Array[String], fileMap: Map[String, File],
+                                       classPath: Array[String], systemConfig: Config,
+                                       userConfigFile: Option[File] = None): Boolean = {
     try {
-      val masters = sysConfig.getStringList(Constants.GEARPUMP_CLUSTER_MASTERS).toList.flatMap(Util.parseHostList)
-      val mastersOption = masters.zipWithIndex.map { kv =>
-        val (master, index) = kv
-        s"-D${Constants.GEARPUMP_CLUSTER_MASTERS}.$index=${master.host}:${master.port}"
-      }.toArray
-
-      val hostname = sysConfig.getString(Constants.GEARPUMP_HOSTNAME)
-      var options = Array(
-        s"-D${Constants.GEARPUMP_HOSTNAME}=$hostname",
-        s"-D${PREFER_IPV4}=true"
-      ) ++ mastersOption
-
-      if (userConf.isDefined) {
-        options :+= s"-D${Constants.GEARPUMP_CUSTOM_CONFIG_FILE}=${userConf.get.getPath}"
+      val jar = fileMap.get("jar")
+      if (jar.isEmpty) {
+        throw new IOException("JAR file not supplied")
       }
 
-      val arguments = Array("-jar", jar.getPath) ++ extraArgs
-      val mainClass = AppSubmitter.getClass.getName.dropRight(1)
-      val process = Util.startProcess(options, Util.getCurrentClassPath, mainClass, arguments)
+      val process = Util.startProcess(
+        clusterOptions(systemConfig, userConfigFile),
+        classPath,
+        mainClass,
+        arguments = createFilePathArgArray(fileMap) ++ argsArray
+      )
+
       val retval = process.exitValue()
       if (retval != 0) {
-        throw new IOException(s"Process exit abnormally with code $retval, error summary: ${process.logger.error}")
+        throw new IOException(s"Process exit abnormally with exit code $retval.\n" +
+          s"Error message: ${process.logger.error}")
       }
-      process.logger.output
-        .split(" ").last.toInt
+      true
     } finally {
-      userConf.foreach(_.delete)
-      jar.delete()
+      fileMap.values.foreach(_.delete)
+      if (userConfigFile.isDefined) {
+        userConfigFile.get.delete()
+      }
     }
   }
 
+  /**
+   * Return Java options for gearpump cluster
+   */
+  private def clusterOptions(systemConfig: Config, userConfigFile: Option[File]): Array[String] = {
+    var options = Array(
+      s"-D${Constants.GEARPUMP_HOME}=${systemConfig.getString(Constants.GEARPUMP_HOME)}",
```

Comment:
do we need `"-Djava.rmi.server.hostname=localhost"` ? I see they are set for `StormRunner` in pack